CMakeLists.txt

```diff
@@ -617,8 +617,13 @@ if(WITH_SYSTEM_BOOST)
       INTERFACE_LINK_LIBRARIES ZLIB::ZLIB)
   endif()
 else()
-  set(BOOST_J 1 CACHE STRING
-    "max jobs for Boost build") # override w/-DBOOST_J=<n>
+  include(ProcessorCount)
+  ProcessorCount(N)
+  if((NOT DEFINED BOOST_J) AND (NOT N EQUAL 0))
+    set(BOOST_J ${N} CACHE STRING "detected max jobs for boost build")
+  else()
+    set(BOOST_J 1 CACHE STRING "max jobs for Boost build") # override w/-DBOOST_J=<n>
+  endif()
```

Comment:
why not just define the default value to `N` ? like
```cmake
include(ProcessorCount)
ProcessorCount(DEFAULT_BOOST_JOBS)
if(DEFAULT_BOOST_JOBS EQUAL 0)
  set(DEFAULT_BOOST_JOBS 1)
endif()
set(BOOST_J ${DEFAULT_BOOST_JOBS} CACHE STRING "max jobs for Boost build")
```

couple changes:

* as user-provided `BOOST_J` always overrides the value set using `set(...)` cmake command. there is no need to check for `NOT DEFINED BOOST_J`
* use a better variable name than simply `N`, for better readability

Reply:
Thanks for the review. I wasn't aware of that specific behavior of `set`, but it makes sense to go that route then.